src/components/Button/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { Component } from 'react';
+import { Link } from './index.styles';
+
+interface ButtonProps {
+  link: string;
+  text: string;
+  background?: string;
+  size?: string;
+  target?: string;
+}
+
+export default class Button extends Component<ButtonProps> {
+  render() {
+    const { link, background, size, text, target } = this.props;
+
+    // check if link is external or not
+    if (link.includes('https')) {
+      // return external
+      return (
+        <Link
+          href={link}
+          background={background}
+          size={size}
+          target={target === '_blank' ? '_blank' : '_self'}
```

Comment:
I think typescript didn't like `target={target}` so I had to work around it. 
`interface ButtonProps extends React.HTMLAttributes<HTMLAnchorElement> {` - Witchcraft?   
Ok, how do you know these kind of things `React.HTMLAttributes<HTMLAnchorElement>` exists and how to use them? Haven't gotten that deep into typescript yet.  
I like the `{...otherProps]`-thing. I will try it your way, thank you.

Reply:
If I try to do it exactly like in your example, typescript complains. But if I keep the   `target?: string;` in the interface, it works beautifully.  
I'm just worried that if we pass any arbitrary otherProps it will crash. Yes tested.
![2019-01-10_1306](https://user-images.githubusercontent.com/22402899/50964746-c315fb80-14d8-11e9-9a2c-3648bb077355.png)

So we need to add any new parameters we pass thru to interface anyway, {...otherProps} won't just work on itself.

Reply:
Okay, that works nicely. Now you can pass target & it works, but any other will be caught by the evil typescript monster. I think that is what we want?  
If so, all the changes shouldbe now made.